core/utils.py

```diff
@@ -0,0 +1,33 @@
+import json
+
+import jwt
+
+from django.http        import JsonResponse
+from users.models       import User
+from django.conf        import settings
+
+def LoginDecorator(func):
+    def wrapper(self, request, *args, **kwargs):
+        try:
+            if 'Authorization' not in request.headers:
+                return JsonResponse({'message' : 'NO_PERMISSION'}, status = 401)
```

Comment:
우선 headers에 `Authorization`이 담겨오지 않으면 아래 jwt 부분에서 에러가 나긴 할겁니다.
그래서 굳이 필요할까? 라는 생각이 들기도 합니다.

추가로 이렇게 디테일하게 조건문을 작성해주신거라면 그에 맞게 메세지도 더 디테일한 것이 좋아 보입니다.
```suggestion
            if 'Authorization' not in request.headers:
                return JsonResponse({'message' : 'no authorization'}, status = 401)
```

Reply:
사실 굳이 있어야하나 고민을 많이했는데 없어도 괜찮았군요! 감사합니다👍🏻